### DISCUSSION TITLE
List rules/rectors that were executed

### DISCUSSION THREAD

**Discussion by cweiske** [rectorphp/rector#7164] [Q&A] [answered]
I'm using rector 0.12.23 with [typo3-rector](https://github.com/sabbelasichon/typo3-rector/) which provides pre-defined sets. One rule/rector is not applied, and I want to see if the rule is loaded at all in the set.

I would have thought that `--debug` or `-vvv` would give me this information, but it does not.

How can I quickly find out which rules have been loaded, without inspecting the source code of the set and drilling down deeper into the files?

**Comment by TomasVotruba** [comment, marked as answer]
Hi, we've added rule + file debug output in the latest `dev-main` when you run it with `--debug` option.
Try that on single file. That should give you the full list of run rules.

**Reply by samsonasik** [reply]
@TomasVotruba it seems --debug doesn't show anything, while `-vvv` show it:

**with --debug**

```
bin/rector process src/Config/  --clear-cache --debug

                                                                                                                        
 [OK] Rector is done!                                                                                                                                                                                                                 
````

**with -vvv**

```
bin/rector process src/Config/  --clear-cache -vvv |more
src/Config/Loader/ConfigureCallMergingLoaderFactory.php
[file] src/Config/Loader/ConfigureCallMergingLoaderFactory.php
[rule] Rector\DeadCode\Rector\Node\RemoveNonExistingVarAnnotationRector

[file] src/Config/Loader/ConfigureCallMergingLoaderFactory.php
[rule] Rector\CodingStyle\Rector\Stmt\NewlineAfterStatementRector

[file] src/Config/Loader/ConfigureCallMergingLoaderFactory.php
[rule] Rector\DeadCode\Rector\Node\RemoveNonExistingVarAnnotationRector

[file] src/Config/Loader/ConfigureCallMergingLoaderFactory.php
[rule] Rector\CodingStyle\Rector\Stmt\NewlineAfterStatementRector
```

**Reply by TomasVotruba** [reply]
Hm :thinking:  Could check what is wrong there?

**Reply by samsonasik** [reply]
I will try.

**Reply by samsonasik** [reply]
It seems it because of `parallel` process, when it run on non-parallel, it working ok.

**Reply by samsonasik** [reply]
https://github.com/rectorphp/rector-src/pull/2307

**Comment by kix** [comment]
What about adding an option that would allow listing all the applied rules after a run? This way, I can tell when a big refactor is safe or not, based on the rules that were executed. Some are risky, some are not: it would be great to be able to see that,

**Reply by TomasVotruba** [reply]
I think this was added in one of last ~3 releases in some CLI `--option`. Check the release notes.

**Reply by kix** [reply]
Right, `--rules-summary`, thanks @TomasVotruba and @carlos-granados for the implementation: https://github.com/rectorphp/rector/releases/tag/2.4.1
